File: 1656-design-an-ordered-stream/1656-design-an-ordered-stream.py
class OrderedStream:

    def __init__(self, n: int):
        self.stream = [None] * (n + 2)
        self.prev = 1

    def insert(self, idKey: int, value: str) -> List[str]:
        self.stream[idKey] = value
        if idKey == self.prev:
            while self.stream[self.prev]:
                self.prev += 1
        return self.stream[idKey:self.prev]


# Your OrderedStream object will be instantiated and called as such:
# obj = OrderedStream(n)
# param_1 = obj.insert(idKey,value)

File: 1656-design-an-ordered-stream/1656-design-an-ordered-stream.js
/**
 * @param {number} n
 */
var OrderedStream = function(n) {
  this.prev = 1;
  this.stream = [];
};

/** 
 * @param {number} idKey 
 * @param {string} value
 * @return {string[]}
 */
OrderedStream.prototype.insert = function(idKey, value) {
  this.stream[idKey] = value;
  if (idKey === this.prev)
    while (this.stream[this.prev]) this.prev++;
  
  return this.stream.slice(idKey, this.prev);
};

/** 
 * Your OrderedStream object will be instantiated and called as such:
 * var obj = new OrderedStream(n)
 * var param_1 = obj.insert(idKey,value)
 */